hoomd/md/NeighborListGPU.cc

```diff
@@ -185,8 +185,6 @@ void NeighborListGPU::filterNlist()
 //! Update the exclusion list on the GPU
 void NeighborListGPU::updateExListIdx()
     {
-    assert(!m_n_particles_changed);
-
```

Comment:
Add this assert back to the code. This method assumes that the number of particles allocated has not changed. #1059 and #1116 fix the Neighbor list code so that it updates these arrays properly and runs without asserts in debug mode.